frida/core.py

```diff
@@ -198,22 +229,46 @@ class ScriptPayloadMessage(TypedDict):
 ScriptDestroyedCallback = Callable[[], None]
 
 
+class RPCException(Exception):
+    """
+    Wraps remote errors from the script RPC
+    """
+
+    def __str__(self) -> str:
+        return str(self.args[2]) if len(self.args) >= 3 else str(self.args[0])
+
+
 class Script:
     def __init__(self, impl: _frida.Script) -> None:
-        self.exports = ScriptExports(self)
+        self.exports_sync = ScriptExportsSync(self)
+        self.exports_async = ScriptExportsAsync(self)
 
         self._impl = impl
 
         self._on_message_callbacks: List[ScriptMessageCallback] = []
         self._log_handler: Callable[[str, str], None] = self.default_log_handler
 
-        self._pending: Dict[int, Callable[..., Any]] = {}
+        self._pending: Dict[
+            int, Callable[[Optional[Any], Optional[Union[RPCException, _frida.InvalidOperationError]]], None]
+        ] = {}
         self._next_request_id = 1
         self._cond = threading.Condition()
 
         impl.on("destroyed", self._on_destroyed)
         impl.on("message", self._on_message)
 
+    @property
+    def exports(self) -> ScriptExportsSync:
+        """
+        The old way of retrieving the synchronous exports caller
+        """
+
+        warnings.warn(
```

Comment:
You should add `stacklevel=2`.
this makes the warning show you the place where you used the deprecated function (in this case `Script.export`) instead of where you used `warnings.warn`